assembly/overworld_scripts/routes/route_8.s

```diff
@@ -0,0 +1,196 @@
+.thumb
+.align 2
+
+.include "../xse_commands.s"
+.include "../xse_defines.s"
+.include "../asm_defines.s"
+
+.global MapScript_Route8
+MapScript_Route8:
+    mapscript MAP_SCRIPT_ON_LOAD MapEntryScript_Route8_SetWeather
+    .byte MAP_SCRIPT_TERMIN
+
+MapEntryScript_Route8_SetWeather:
+    @ Check player position, to see if they spawned from the north gate or fisherman's hut
+    getplayerpos 0x4000 0x4001
+    compare 0x4001 0xB @ Either north gate (0xA) or Fisherman's hut (0xB), no need to set weather
+    if greaterthan _goto TileScript_Route8_SetWeather    
+    end
+
+.global TileScript_Route8_ClearWeather
+TileScript_Route8_ClearWeather:
+    setvar 0x400F 0x0 @ Enable weather setting again
+    call SetWeatherClear
+    End
+
+.global TileScript_Route8_SetWeather
+TileScript_Route8_SetWeather:
+    compare 0x400F 0x1
+    if equal _goto End
+    setvar 0x400F 0x1 @ Prevent future weather until var is cleared
+    call SetWeatherFog
+    random 0x64 @ Between 0 and 100
+    compare LASTRESULT 0xF @ "15"
+    if lessorequal _goto SetWeatherMistyAndEnd
+    compare LASTRESULT 0x2D @ "45"
+    if greaterorequal _goto SetWeatherRainyAndEnd
+    goto TileScript_Route8_ClearWeather
+
+SetWeatherMistyAndEnd:
+    call SetWeatherFog
+    goto End
```

Comment:
End not end?

Reply:
Good point, I made the change.

End is a helper function that includes release and end statements, but there's nothing locked to be released here. In the end it consumes more space for the call to a function (5 bytes) to just ending (1)